website/src/components/Survey/TrackedTextarea.tsx

```diff
@@ -0,0 +1,35 @@
+import { Progress, Stack, Textarea, TextareaProps } from "@chakra-ui/react";
+
+interface TrackedTextboxProps {
+  text: string;
+  thresholds: {
+    low: number;
+    medium: number;
+    goal: number;
+  };
+  textareaProps?: TextareaProps;
+  onTextChange: (event: React.ChangeEvent<HTMLTextAreaElement>) => void;
+}
+
+export const TrackedTextarea = (props: TrackedTextboxProps) => {
```

Comment:
I like how this is its own component.  Later one we could add more to make the writing experience better.

A small nit I might fix in the future is making this a `forwardRef` so that we can avoid having a state variable.